ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/BufferedWriteChannel.java

```diff
@@ -45,16 +50,18 @@ static BufferedWriteChannel open(File file, boolean append, ByteBuffer buffer) t
       fc.truncate(0);
     }
     Preconditions.assertSame(fc.size(), fc.position(), "fc.position");
-    return new BufferedWriteChannel(fc, buffer);
+    return new BufferedWriteChannel(fc, buffer, closeCallback);
   }
 
   private final FileChannel fileChannel;
   private final ByteBuffer writeBuffer;
   private boolean forced = true;
+  private final Supplier<CompletableFuture> callback;
```

Comment:
Please specify the type parameter.  Otherwise, there is a "Raw use of parameterized class 'CompletableFuture'" warning.